### PR TITLE
enables gcs staging databricks

### LINKED ISSUE #1902: Allow GCS staging buckets for Databricks destination
### Feature description

The Databricks destination stages data in an external bucket before copying the data into Delta Lake. I'm working on GCP and need to use a GCS bucket for staging the data. I was able to get this working with a forked version of the repo.

### Are you a dlt user?

Yes, I run dlt in production.

### Use case

My company is migrating to Databricks on GCP. We have a dozen dlt pipelines in production that will need to be pointed to Databricks. 

### Proposed solution

Modify `dlt/destinations/impl/databricks/databricks.py` to allow a `gs` bucket_scheme makes its way to https://github.com/dlt-hub/dlt/blob/854905fb56576bc608b01b6b047208df888160a7/dlt/destinations/impl/databricks/databricks.py#L83
If there isn't a storage credential, still throw an exception saying GCS buckets do not work with temporary credentials.

### Related issues

_No response_

### PR DESCRIPTION
**Pull request by rudolfix**
### Description
fixes #1902

### CODE REVIEW

File: docs/website/docs/dlt-ecosystem/destinations/databricks.md [hunk 1]
```diff
@@ -141,7 +141,7 @@ The `jsonl` format has some limitations when used with Databricks:
 
 ## Staging support
 
-Databricks supports both Amazon S3 and Azure Blob Storage as staging locations. `dlt` will upload files in `parquet` format to the staging location and will instruct Databricks to load data from there.
+Databricks supports both Amazon S3, Azure Blob Storage and Google GCS as staging locations. `dlt` will upload files in `parquet` format to the staging location and will instruct Databricks to load data from there.
```
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/destinations/databricks.md:144]:
```suggestion
Databricks supports both Amazon S3, Azure Blob Storage and Google Cloud Storage as staging locations. `dlt` will upload files in `parquet` format to the staging location and will instruct Databricks to load data from there.
```

File: docs/website/docs/dlt-ecosystem/destinations/databricks.md [hunk 2]
```diff
@@ -187,6 +187,11 @@ pipeline = dlt.pipeline(
 
 ```
 
+### Databricks and Google Cloud Storage
+
+In order to load from GCS stage you must set-up the credentials via **named credential**. See below. Databricks does not allow to pass Google Credentials
```
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/destinations/databricks.md:192]:
```suggestion
In order to load from Google Cloud Storage stage you must set-up the credentials via **named credential**. See below. Databricks does not allow to pass Google Credentials
```

File: tests/load/pipeline/test_databricks_pipeline.py [hunk 3]
```diff
@@ -90,3 +95,68 @@ def test_databricks_external_location(destination_config: DestinationTestConfigu
...
+        staging=stage,
+    )
+    info = pipeline.run([1, 2, 3], table_name="digits", **destination_config.run_kwargs)
+    assert info.has_failed_jobs is True
+    assert (
+        "You need to use Databricks named credential or external location"
+        in pipeline.list_failed_jobs_in_package(info.loads_ids[0])[0].failed_message
+    )
+
+    # should fail on internal config error as external location is not configured
```
Comment (burnash) [inline tests/load/pipeline/test_databricks_pipeline.py:135]: Maybe remove the commented code?

**Review by burnash — APPROVED** [review]

File: tests/load/pipeline/test_databricks_pipeline.py (hunk 3 above)
Comment (rudolfix) [reply tests/load/pipeline/test_databricks_pipeline.py:135]: OK I'll do that when test finish

**Pull request merged**